### ISSUE TITLE
Improve email debugging with enabled dev tools

### ISSUE THREAD

**Issue by dtdesign** [WoltLab/WCF#2501]
Debugging emails in its current state is a pain, figuring out issues related to the whole email creation and transport is handled asynchronously and with all sorts of suppressed error message.

When the dev tools are enabled:
* [x] All items added to the background queue are processed immediately in the same process (synchronous execution).
* [x] Exceptions are not suppressed, regardless of the source, including but not limited to temporary errors returned from a remote mail server.

The bottom line is that the mail has been sent (or written to a log file) at the end of the current request, in case of any error, the occurred exception is not suppressed.
